### ISSUE TITLE
Slowdown/fps drop after editing or adding code in 0.4.0 prelease (1-3)

### ISSUE THREAD

**Issue by Jesright73** [krzys-h/UndertaleModTool#504]
Using UMT to modify a fan game known as AM2R 1.5.2.

Whenever any code is edited or added using UMT prelease 0.4.0, slowdown is introduced into the game. The fps drop is very noticeable in rooms with many lighting effects.

Committing the same code edits in UMT 0.3.5.8 does not result in slowdowns.

**Comment by Archie-osu** [comment]
Hi, can you post a sample of the code you're committing?

**Comment by Jesright73** [comment]
The steps i've been taking to reproduce it are as follows:
1: right-click `Code`, select `Add`
2: name new code as anything, i used `testobject` for my example. i select `Change only this one value` when the string prompt appears.
3: type (or paste) _any_ code into the code entry box. The code i typed for thiswas `image_speed = 1`, though it doesn't really matter as long as there is a line of code
4:click outside the code box to commit
5: save the data.win

**Comment by Jesright73** [comment]
here is the assembly of the code i am adding to the file

; WARNING: Missing code locals, possibly due to unsupported bytecode version or a brand new code entry.

:[0]
pushi.e 1
pop.v.i self.image_speed

:[end]

**Comment by Archie-osu** [comment]
Oh yeah that looks normal.
Does this happen on other games, or is AM2R the only one?

**Comment by Jesright73** [comment]
That i haven't tested yet. I don't have Undertale, but i will take a look at what i do have that could be used for further testing.
Something is very strange about this case. 
Editing version 1.4.3 of am2r with prelease 0.4.0 does not result in the slowdowns, while any later version of am2r it does; 
and as stated above, editing am2r 1.5.2 with UMT 0.3.5.8 does not result in slowdown either.

 It is only when using UMT prelease 0.4.0 to edit code in AM2R 1.5.2 that the slowdown occurs.

**Comment by Archie-osu** [comment]
Hmm, might be the case that something changed in the newer version (theory courtesy of @BenjaminUrquhart).
Though probably the best course of action right now would be to use 0.3.5.8 in the meantime, while we look into it.

**Comment by Kneesnap** [comment]
It might help if some kind of video was shown. Information like which game,
which code entry, which part of the code, etc, which may be tedious to
describe will all be easily determinable if there is a video showcasing the
bug.

**Comment by Archie-osu** [comment]
He's creating his own code entry in AM2R 1.5.2, adding only one line.
In my opinion the information provided is enough, but if you reckon it'll help, go for it.

**Comment by Jesright73** [comment]
i opened the file with 0.3.5.8 that was modified with 0.4.0
Selected a code entry that i wrote using 0.4.0, and then selected the `Decompiled` tab.
This error appeared 
![image](https://user-images.githubusercontent.com/79653967/128024767-b1a59805-cefd-41ef-9661-908065f433be.png)

**Comment by Jesright73** [comment]
As i have investigated further, i have found something that might be a useful clue.
i have opened the data.win of AM2R multitroid (a mod of am2r 1.5.2) in UMT prelease 0.4.0. I opened the code at the bottom of the list, modified a single value, committed, then reverted it to the original value.
i then opened the two data.win files in Winmerge. The lines in the modified data.win are not aligning with the original, even though nothing except for a single character in code was changed and reverted.
i have a screenshot of the first 30-some lines of code, and am also uploading the two data.win files if anyone here is interested in taking a closer look.
![image](https://user-images.githubusercontent.com/79653967/128106685-da50ac8b-79a8-4a43-ae93-8107e5493693.png)

Link to the data.win files
https://drive.google.com/file/d/1yPMg3VsbtrqlihmBfI8Ca-wmcwfhum1m/view?usp=sharing

**Comment by Grossley** [comment]
You are using the wrong comparison mode, that is garbage output.

**Comment by Kneesnap** [comment]
This is normal for any change in a data.win

**Comment by Grossley** [comment]
> The lines in the modified data.win are not aligning with the original, even though nothing except for a single character in code was changed and reverted.

Also this is normal

**Comment by Grossley** [comment]
> This is normal for any change in a data.win
> […](#)

This is true for code changes, however many changes are truly reversible, code changes are not one of them.

**Comment by Jesright73** [comment]
Is it normal for the License MD5 and License CRC32 to be different when opening the same data.win with two different versions of UMT? Exact same file, left side is 0.3.8.5, right side is prelease_3 0.4.0
![image](https://user-images.githubusercontent.com/79653967/129279427-df8c42ec-d49a-4250-87b2-e24eaed683a0.png)

**Comment by Grossley** [comment]
> Is it normal for the License MD5 and License CRC32 to be different when opening the same data.win with two different versions of UMT? Exact same file, left side is 0.3.8.5, right side is prelease_3 0.4.0
> ![image](https://user-images.githubusercontent.com/79653967/129279427-df8c42ec-d49a-4250-87b2-e24eaed683a0.png)

This is unrelated, please do not use this issue for this matter @Jesright73 . Also, this is the same data but reordered. I recall reordering these in the interim as the order presented was improper in 0.3.5.8 but I do not know if the data in the screenshots are reordered properly. Please make a new issue if this is something you wish to be investigated.

**Comment by Grossley** [comment]
I think one of these three commits may be responsible for the issues in #504

https://github.com/krzys-h/UndertaleModTool/commit/eb4613fefd48106568cc4e8b68b897083d4fbb46 Works, has no slowdown at all.
https://github.com/krzys-h/UndertaleModTool/commit/431f5f4f3a7bc03ce71d3e9bf655758c7c5f5348 Game just crashes out of the blue after `Run_Start`
https://github.com/krzys-h/UndertaleModTool/commit/2abae50786f696db75c8687b4c106737f2bf5712 Doesn't crash, has slowdown issues

Help determining how these changes interact with AM2R 1.5.2+ would be helpful

**Comment by Miepee** [comment]
That info is actually incorrect.
`288db43` until `431f5f4` all crash after `Run_Start`
`afdd618` is the first commit that runs again, but with the slowdown issues.
